eo-maven-plugin/src/test/java/org/eolang/maven/testapi/FakeMaven.java

```diff
@@ -0,0 +1,195 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.testapi;
+
+import com.yegor256.tojos.Tojo;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.maven.plugin.AbstractMojo;
+import org.eolang.maven.AssembleMojo;
+import org.eolang.maven.Catalogs;
+import org.eolang.maven.Home;
+import org.eolang.maven.Moja;
+
+/**
+ * Fake maven workspace that executes Mojos in order to test
+ * their behaviour and results.
+ * @since 0.28.12
+ * @todo #1417:90min Make `FakeMaven.exec()` return a `HashMap` with all
+ *  files created in the directory and their relative names. Then, we can assert on this hash map.
+ */
+public final class FakeMaven {
+
+    /**
+     * Default eo program id.
+     */
+    private static final String PROGRAM_ID = "foo.x.main";
+
+    /**
+     * Default eo program path.
+     */
+    private static final String PROGRAM_PATH = "foo/x/main.eo";
+
+    /**
+     * Default eo-foreign.csv file format.
+     */
+    private static final String FOREIGN_FORMAT = "csv";
+
+    /**
+     * Default eo-foreign.csv file path.
+     */
+    private static final Path FOREIGN_PATH = Paths.get("eo-foreign.csv");
+
+    /**
+     * Test workspace where we place all programs, files, compilation results, etc.
+     */
+    private final Home workspace;
+
+    /**
+     * Mojos params.
+     */
+    private final Map<String, Object> params;
+
+    /**
+     * Attributes for eo.foreign.*.
+     */
+    private final Map<String, Object> attributes;
+
+    /**
+     * Path to a program in workspace.
+     */
+    private Path prog;
+
+    /**
+     * The main constructor.
+     *
+     * @param workspace Test temporary directory.
+     */
+    public FakeMaven(final Path workspace) {
+        this.workspace = new Home(workspace);
+        this.params = this.defaultParams();
```

Comment:
@volodya-lombrozo to be honest, this looks wrong. Ctors [must be](https://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html) code-free. When you have a need to put some code into ctor, your design is in trouble. Here, I would suggest to make a new method `FakeMaven.withDefaults()`. The user must call it if they need defaults.

Reply:
@yegor256 fixed.